ui/src/app/zones/views/ZonesList/ZonesList.tsx

```diff
@@ -1,13 +1,29 @@
+import { useEffect } from "react";
+
+import { useDispatch, useSelector } from "react-redux";
+
 import ZonesListHeader from "../ZonesListHeader";
+import ZonesListTable from "../ZonesListTable";
 
 import Section from "app/base/components/Section";
+import { actions } from "app/store/zone";
+import zoneSelectors from "app/store/zone/selectors";
 
 const ZonesList = (): JSX.Element => {
+  const dispatch = useDispatch();
+  const zonesCount = useSelector(zoneSelectors.count);
+
+  useEffect(() => {
+    dispatch(actions.fetch());
+  }, [dispatch]);
+
   return (
     <Section
       header={<ZonesListHeader />}
       headerClassName="u-no-padding--bottom"
-    />
+    >
+      {zonesCount > 0 && <ZonesListTable />}
```

Comment:
Here you might want to show a spinner while the data is loading. You could get the loading state using `const loading = useSelector(zoneSelectors.loading);` and then you could display the Spinner component from react-components.

You might also want to add a message if the zones have loaded but there are no zones returned.